app/src/main/java/org/simple/clinic/instantsearch/InstantSearchEffectHandler.kt

```diff
@@ -73,6 +76,22 @@ class InstantSearchEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun checkIfPatientAlreadyHasAnExistingNHID(): ObservableTransformer<CheckIfPatientAlreadyHasAnExistingNHID, InstantSearchEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulers.computation())
+          .flatMap { patientRepository.patientProfile(it.patientId) }
+          .map { it.get().withoutDeletedBusinessIds() }
+          .map { checkIfPatientProfileHasNationalHealthId(it) }
+    }
+  }
+
+  private fun checkIfPatientProfileHasNationalHealthId(it: PatientProfile) = if (it.hasNationalHealthID) {
```

Comment:
Can you rename the method parameter please? When you use the IDE extract function, it does not do that good a job of selecting the name.